lib/exercism/user.rb

```diff
@@ -52,21 +52,21 @@ def access?(problem)
   end
 
   # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
-  def self.from_github(id, username, email, avatar_url, joined_as=nil)
-    user = User.where(github_id: id).first
+  def self.from_github(github_user, joined_as=nil)
```

Comment:
Or maybe `from_authentication_data`, since it takes both a GitHubUser and the `joined_as` string.